redis/config.go

```diff
@@ -45,6 +45,7 @@ type Config struct {
 	Host       string     `yaml:"host"`
 	Port       int        `yaml:"port"`
 	Password   string     `yaml:"password"`
+	Database   int        `yaml:"database" default:"0"`
```

Comment:
Please validate the decoded value (`0 =< n < 16`) and reject it if it's invalid, instead of crashing later at runtime.
```bash
2024-08-01T14:41:00.154+0200    FATAL   main    ERR DB index is out of range
can't connect to Redis
main.run
        /Users/yhabteab/Workspace/go/icingadb/cmd/icingadb/main.go:82
main.main
        /Users/yhabteab/Workspace/go/icingadb/cmd/icingadb/main.go:37
runtime.main
        /opt/homebrew/opt/go/libexec/src/runtime/proc.go:271
runtime.goexit
        /opt/homebrew/opt/go/libexec/src/runtime/asm_arm64.s:1222
exit status 1
```

Reply:
As I have written in the [referenced issue](https://github.com/Icinga/icingadb/issues/676#issuecomment-1918641933), "there is no hard limit of sixteen databases. This is just the [default in the configuration](https://raw.githubusercontent.com/redis/redis/7.2/redis.conf)." Thus, I would just let it crash if the user decides to use something outside of the configured Redis scope.